examples/flutter_gallery/lib/demo/animation_demo.dart

```diff
@@ -0,0 +1,439 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+
+enum _DragTarget {
+  start,
+  end
+}
+
+// How close a drag's start position must be to the target point. This is
+// a distance squared.
+const double _kTargetSlop = 2500.0;
+
+// Used by the Painter classes.
+const double _kPointRadius = 6.0;
+
+class _DragHandler extends Drag {
+  _DragHandler(this.updateCallback, this.cancelCallback, this.endCallback);
+
+  final GestureDragUpdateCallback updateCallback;
+  final GestureDragCancelCallback cancelCallback;
+  final GestureDragEndCallback endCallback;
+
+  @override
+  void update(DragUpdateDetails details)  {
+    updateCallback(details);
+  }
+
+  @override
+  void cancel()  {
+    cancelCallback();
+  }
+
+  @override
+  void end(DragEndDetails details)  {
+    endCallback(details);
+  }
+}
+
+class _IgnoreDrag extends Drag {
+}
+
+class _PointDemoPainter extends CustomPainter {
+  _PointDemoPainter({
+    Animation<double> repaint,
+    this.arc
+  }) : _repaint = repaint, super(repaint: repaint);
+
+  final MaterialPointArcTween arc;
+  Animation<double> _repaint;
+
+  void drawPoint(Canvas canvas, Point point, Color color) {
+    final Paint paint = new Paint()
+      ..color = color.withOpacity(0.25)
+      ..style = PaintingStyle.fill;
+    canvas.drawCircle(point, _kPointRadius, paint);
+    paint
+      ..color = color
+      ..style = PaintingStyle.stroke
+      ..strokeWidth = 2.0;
+    canvas.drawCircle(point, _kPointRadius + 1.0, paint);
+  }
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    final Paint paint = new Paint();
+
+    if (arc.center != null)
+      drawPoint(canvas, arc.center, Colors.blue[400]);
+
+    paint
+      ..color = Colors.green[500].withOpacity(0.25)
+      ..strokeWidth = 4.0
+      ..style = PaintingStyle.stroke;
+    if (arc.center != null && arc.radius != null)
+      canvas.drawCircle(arc.center, arc.radius, paint);
+    else
+      canvas.drawLine(arc.begin, arc.end, paint);
+
+    drawPoint(canvas, arc.begin, Colors.green[500]);
+    drawPoint(canvas, arc.end, Colors.red[500]);
+
+    paint
+      ..color = Colors.green[500]
+      ..style = PaintingStyle.fill;
+    canvas.drawCircle(arc.lerp(_repaint.value), _kPointRadius, paint);
+  }
+
+  @override
+  bool hitTest(Point position) {
+    return (arc.begin - position).distanceSquared < _kTargetSlop
+        || (arc.end - position).distanceSquared < _kTargetSlop;
+  }
+
+  @override
+  bool shouldRepaint(_PointDemoPainter oldPainter) => true;
```

Comment:
I have to admit that I was returning true out of sloth. I've added `operator ==` methods to the arc tween classes so that I can check if the painter's arc has changed here.